crates/core/src/operations/transaction/state.rs

```diff
@@ -214,6 +214,21 @@ impl<'a> PruningStatistics for AddContainer<'a> {
         ScalarValue::iter_to_array(values).ok()
     }
 
+    /// return the number of rows for the named column in each container
+    /// as an `Option<UInt64Array>`.
+    ///
+    /// Note: the returned array must contain `num_containers()` rows
+    fn row_counts(&self, _column: &Column) -> Option<ArrayRef> {
+        let values = self.inner.iter().map(|add| {
```

Comment:
@ion-elgreco I'm assuming we have the same row count for each column, provided by the stats.

Reply:
Yeah total row count should be on file level